test/e2e/clusters/clusters.go

```diff
@@ -0,0 +1,256 @@
+	config, err := GetConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	clientSet, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return clientSet, nil
+}
+
+// createKindCluster creates a KinD cluster given the path to use to create a kubeconfig file.
+// It returns an error if something went wrong.
+func createKindCluster(kubeconfigFile string) error {
+	provider := kindCluster.NewProvider(kindCluster.ProviderWithLogger(kindcmd.NewLogger()))
+	err := provider.Create(
+		kindClusterName,
+		kindCluster.CreateWithNodeImage(""),
+		kindCluster.CreateWithRetain(false),
+		kindCluster.CreateWithWaitForReady(time.Duration(0)),
+		kindCluster.CreateWithKubeconfigPath(kubeconfigFile),
+		kindCluster.CreateWithDisplayUsage(false))
+	if err != nil {
+		return err
+	}
+	// The cluster needs a bit more time to become healthy
+	err = waitForHealthyCluster()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// deleteKindCluster deletes the KinD cluster that was created at the beginning of the test suite, and removes the
+// cluster configuration from the provided Kubeconfig file. It does not delete the file itself.
+func deleteKindCluster(kubeconfigFile string) error {
+	provider := kindCluster.NewProvider(kindCluster.ProviderWithLogger(kindcmd.NewLogger()))
+	err := provider.Delete(kindClusterName, kubeconfigFile)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// createK3dClusterUsingCurrentKubeconfig creates a K3d cluster. It will automatically modify the currently configured
+// Kubeconfig file and switch the current context to the new cluster. It returns an error if something went wrong.
+func createK3dClusterUsingCurrentKubeconfig() error {
+	// Guide on how to do args and flags:
+	// https://stackoverflow.com/questions/49848898/cobra-how-to-set-flags-programmatically-in-tests/50880663#50880663
+	createClusterCommand := k3dCluster.NewCmdClusterCreate()
+	createClusterCommand.SetArgs([]string{
+		k3dClusterName,
+		"--kubeconfig-update-default=true",
+		"--kubeconfig-switch-context=true",
+		"--wait=true",
+		"--timeout=120s",
+	})
+	err := createClusterCommand.ExecuteContext(context.TODO())
+	if err != nil {
+		return err
+	}
+	// The cluster needs a bit more time to become healthy
+	err = waitForHealthyCluster()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// deleteK3dCluster deletes the K3d cluster that was created at the beginning of the test suite. It uses the currently
+// configured Kubeconfig file (from the KUBECONFIG env var).
+func deleteK3dCluster() error {
+	deleteClusterCommand := k3dCluster.NewCmdClusterDelete()
+	deleteClusterCommand.SetArgs([]string{
+		k3dClusterName,
+	})
+	err := deleteClusterCommand.ExecuteContext(context.TODO())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// waitForHealthyCluster can be used to wait until the cluster is healthy if it needs a little extra time to spin up.
+func waitForHealthyCluster() error {
```

Comment:
Same comments as before, we already have wait logic to wait for a healthy cluster we should be testing, not implementing custom logic to bypass it for tests.

Reply:
Okay then put another way, we should use the existing logic in the codebase to wait for a cluster, I think that's already been agreed in a prior comment.